spec/request/request_for_comments_spec.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'POST /request_for_comments/:rfc_id/report', type: :request do
+  it 'sends an email to let admins know about the report' do
+    user = create(:learner)
+    password = attributes_for(:learner).fetch(:password)
+    login_as(user, password)
```

Comment:
As request specs are to test a specific request (and not the authentication), please stub the login instead of actually making a post request). E.g. with `allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)` in a `before` action.

Reply:
Thanks! The `allow_any_instance_of` approach is already used in CodeOcean and CodeHarbor, so that it alings well. Allow me just one or two remarks regarding the helper:

From my perspective, I usually avoided placing two modules in a single file (as you've just done, `module Authentication` and `module RequestLoginHelper` are both placed in `spec/support/authentication.rb`). While we can introduce the helper as you did :+1:, I would also suggest to use it consistently throughout the specs (all others don't have a helper, but just use this line).

Reply:
Okay! 

My main motivation with the helper was to hide/centralize the use of `allow_any_instance_of` since it is [generally not recommended to use](https://rspec.info/features/3-13/rspec-mocks/old-syntax/any-instance/). I hope I find a better solution for stubbing authorization at some point.

Reply:
`any_instance_of` is indeed smelly, while the cleaner version using `instance_double`s is more verbose. How about tackling that in a follow-up? I could imagine a `shared_context` for stubbing the login.